Godeps/_workspace/src/k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api/types.go

```diff
@@ -84,6 +84,8 @@ type AuthInfo struct {
 	ClientKeyData []byte `json:"client-key-data,omitempty"`
 	// Token is the bearer token for authentication to the kubernetes cluster.
 	Token string `json:"token,omitempty"`
+	// ActAs is the username to act-as.
+	ActAs string `json:"act-as,omitempty"`
```

Comment:
For controllers, I think will want to have just one client object, and then be able to select the role to act as for each request? That is:

```go
for _, job := range(jobs) {
   if needAnotherPod(job) {
     pod := anotherPod(job)
     actAs := userForControllerToActAs(job)
     client.ActingAs(actAs).Pods(job.spec.template.Namespace).Create(pod)
}
```
     
Or were you thinking of the controller having many clients?